### ISSUE TITLE
Release individual crates upon merge into master

### ISSUE THREAD

**Issue by gnunicorn** [paritytech/substrate#10166]
Right now, the CI job is releasing creates when it finds a `ci-release`-tag on the commit. However, we want the job to run on every merge into master and release all crates where versions have been changed compared to what is released on crates.io at the moment. From a technical perspective this only [requires switching the unleash job](https://github.com/paritytech/substrate/blob/26e5f5f526ced19dfe649010e8f19e2c3f957b12/.gitlab-ci.yml#L870-L882) to run on `master`, but before we can do that we have to consider a few aspects:

- the unleash-check-CI job is currently disabled (I don't know why) but _must_ run the release job, at least on the PR that intends to bump versions, to ensure things aren't breaking (ideally on every PR to make sure we've catch bugs early and not only when we intend to release it)
- can we enforce that changelogs are not only kept but upgrade instructions from the `unreleased`-section are moved properly? Maybe through a helper job and a flag on `cargo unleash` that checks the title isn't present upon release?
- can we provide helpers to the developers to know which bump they have to do (MAJOR, MINOR, etc)
- what is our tagging-story for these releases? Do we want to create a tag per each crate-x.x.x-combo? Would make it easier to find chances to apply hot-fixes, but also pretty messy huge list of tags in this repo?
- How do we prevent upgrade-fatigue? How about upgrade-incompatibilities? If we release very often, maybe even bumping major versions, this requires downstream tooling to upgrade at least their `Cargo.toml` as well. Can we help with dependabot-setups to make sure everyone is on latest? How do we deal with the problem that some downstream tooling might needs a while to upgrade to latest MAJOR while others does and the user that wants to combine both ends up in an unrecoverable situation (cargo doesn't know `semver version ranges`)?

**Comment by gnunicorn** [comment]
> the unleash-check-CI job is currently disabled (I don't know why) but must run the release job, at least on the PR that intends to bump versions, to ensure things aren't breaking (ideally on every PR to make sure we've catch bugs early and not only when we intend to release it)

Digging into the history of this, it seems it was temporarily deactivated to wait for an external fix that has been merged already Sept 15th – just no one re-activated the jobs properly. Gonna give it a try and see if this is a quick one to sort out.
